doc/architecture/04solution.md

```diff
@@ -7,20 +7,29 @@ PermaplanT has a classical frontend/backend architecture:
 
 ## Type Safety
 
-Types are shared whenever possible:
+We use specific types (prefer enum over int over string etc.) and share types whenever possible:
 
-- `Diesel` is used to share types between the database and backend code
-- `typeshare` is used to share types between Rust, API and Typescript
+- [Diesel](https://diesel.rs/) is used to share types between the database and backend code.
+  We use type-safe queries (and not raw queries), whenever possible.
+- [typeshare](https://crates.io/crates/typeshare) is used to share types between Rust, API and Typescript.
+
+## Validation
+
+- The frontend should validate data as early as possible, usually during input using [React Hook Form](https://react-hook-form.com/).
+- We don't need to validate data in the backend.
```

Comment:
I think we should also validate important input in the backend. 
Type validation is done by actix web, but some constraints could be bypassed by just validating in the frontend.
For example, payment data or data associated with gamification achievements.

Reply:
Yes, I agree f76c19312f1b390a174737cf079701e3026118f0